### PR TITLE
H-3045: Update Discord references

### PR DESCRIPTION
**Pull request by vilkinsons**
## 🌟 What is the purpose of this PR?

This PR updates our docs and various pages on the _hash.dev_ and _hash.ai_ sites, as well as HASH App, to remove links to Discord.

Further work has been undertaken in separate PRs:

- [x] **In the public Block Protocol monorepo:** to remove Discord references and components from the BP site and infra (https://github.com/blockprotocol/blockprotocol/pull/1396)
- [x] **In our internal site repo:** to remove Discord-related components from internal HAI site (https://github.com/hashintel/internal-gitstart/pull/108)
- [x] **In our public `labs` and internal `internal` monorepos:** to remove Discord references there (https://github.com/hashintel/labs/pull/57 and https://github.com/hashintel/internal/pull/155).

Where possible, Discord references should be changed to the BP / HASH GitHub Issues and/or Discussions pages as appropriate, on GitHub. In other cases they should be removed entirely.

**Out of scope of all of these PRs, and for future follow-up (in backlog, not on roadmap):**

- Updating Discord icon/components in hCore (there are also some references to Slack in the codebase, still, it seems - which Discord previously replaced).

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -16,11 +15,9 @@ Thanks for taking the time to contribute! 🎉 We've established a set of [commu
 
 If you'd like to make a significant change or re-architecture to this repository or any project within, please first open a [discussion] or create an [issue] to get feedback before spending too much time.
 
-We also have:
+We also have a developer website at [hash.dev], containing developer tutorials, guides and other resources.
 
-- A community [Discord] server where you can ask questions about contributing, and obtain helpful advice
-- A developer website at [hash.dev], containing developer tutorials, guides and other resources
-- A design website at [hash.design] containing our Storybook, brand and style guildelines
+_In the future, we also intend to launch a design website at [hash.design], containing our Storybook, brand and style guildelines. This remains incomplete and currently provides only limited value._
```
Comment (CiaranMn) [inline .github/CONTRIBUTING.md:21]: Not sure we need this
```suggestion
```

File: apps/hashdotdev/src/pages/docs/[[...docs-slug]].page.tsx [hunk 2]
```diff
@@ -260,7 +258,7 @@ const DocsPage: NextPageWithLayout<DocsPageProps> = ({
             px: 1,
           }}
         >
-          Chat to us on Discord
+          Manage API keys
```
Comment (CiaranMn) [inline apps/hashdotdev/src/pages/docs/[[...docs-slug]].page.tsx:261]: Perhaps 'Sign up' or similar?

**Review by CiaranMn — COMMENTED** [review]
Code looks good! A couple of small comments on copy

File: apps/hashdotdev/src/pages/docs/[[...docs-slug]].page.tsx (hunk 2 above)
Comment (vilkinsons) [reply apps/hashdotdev/src/pages/docs/[[...docs-slug]].page.tsx:261]:
```suggestion
          Launch HASH
```

**Pull request merged**